libraries/rush-lib/src/logic/operations/OperationExecutionManager.ts

```diff
@@ -103,7 +98,12 @@ export class OperationExecutionManager {
     this._beforeExecuteOperation = beforeExecuteOperation;
     this._afterExecuteOperation = afterExecuteOperation;
     this._beforeExecuteOperations = beforeExecuteOperations;
-    this._onOperationStatusChanged = onOperationStatusChanged;
+    this._onOperationStatusChanged = (record: OperationExecutionRecord) => {
+      if (record.isRemoteExecuting) {
+        this._executionQueue.assignOperations();
+      }
```

Comment:
Strictly speaking, the two status values for which we should invoke `assignOperations()` are `Ready` and `RemoteExecutingPossiblyComplete` (i.e. the statuses that make an operation become a candidate for execution).